File: 프로그래머스/unrated/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.js
// 가장 가까운 같은 글자

function solution(s) {
  const stringObject = {};
  const answer = [];

  for (let i = 0; i < s.length; i++) {
    if (stringObject[s[i]] === 0 || stringObject[s[i]]) {
      // 객체에 해당 키가 있으면
      answer[i] = i - stringObject[s[i]];
      stringObject[s[i]] = i;
    } else {
      // 객체에 해당 키가 없으면
      answer[i] = -1;
      stringObject[s[i]] = i;
    }
  }
  return answer;
}

File: 프로그래머스/unrated/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.py
def solution(s):
    answer = []
    obj = {}
    
    for index, string in enumerate(s):
        if string not in obj:
            answer.append(-1)
            obj[string] = index
        else:
            answer.append(index - obj[string])
            obj[string] = index
    return answer